### ISSUE TITLE
MQTT: acknowledged delivery

### ISSUE THREAD

**Issue by oberstet** [crossbario/crossbar#889]
WAMP and Crossbar.io has (optional) acknowledged publications. WAMP currently lacks acknowledged delivery of events (eg there is a WAMP EVENT message, but there is not response message to the former).

In it's simplest incarnation, we would need a WAMP message

    [EVENT_PROCESSED, EVENT.Publication|id]

that acknowledges the reception of an `EVENT` that was to be delivered with acknowledged delivery.

This requires work on WAMP and AutobahnPython first.

**Comment by hawkowl** [comment]
I am guessing this would go into https://raw.githubusercontent.com/wamp-proto/wamp-proto/master/rfc/text/basic/bp_pubsub_publishing_and_events.md ? Or would this be an 'advanced profile' item?

**Comment by oberstet** [comment]
This should be part of the WAMP Advanced Profile.

**Comment by hawkowl** [comment]
Right. Are we also going to have exactly-once QoS in WAMP, too? Since this would do QoS1 either way, but not 2.

**Comment by oberstet** [comment]
Yes, this is only about QoS 1 (in MQTT terms).

> QoS 2 is a whole different kind of beast .. need to see where this goes.
